LSP.sc

```diff
@@ -105,6 +105,29 @@ LSPConnection {
         // @TODO Is this the only "default" provider we want?
         this.addProvider(InitializeProvider(this, {}));
         
+        \DocumentationProvider.asClass !? { |docProvider|
+            docProvider.registerProvider(Class, { |class|
+                LSPDatabase.classDocs.atFail(class.name, { 
+                    var stream, doc, node;
+                    doc = SCDoc.documents["Classes/"++class.name];
+
+                    if (doc.isUndocumentedClass.not, {
+                        try {
+                            node = SCDoc.parseDoc(doc);
+                        } {
+                            LSPDatabase.classDocs.put(class.name, "");
```

Comment:
[consider] I wonder if its worth moving the caching outside of the provider to `LSPDatabase`? So the code registered with the `DocumentationProvider` is decoupled from the `LSPDatabase` implementation. Then it just has the single responsibility of taking a class, and providing the docs for it?